File: scala/src/test/scala/cucumber/runtime/scala/test/StepDefs.scala
package cucumber.runtime.scala.test

import _root_.cucumber.api.scala._

import cucumber.api.DataTable
import junit.framework.Assert._
import scala.collection.JavaConversions._

object CukesStepDefinitions extends ScalaDsl with EN {

  var calorieCount = 0.0

  Given("""^I have (\d+) "([^"]*)" in my belly$"""){ (howMany:Int, what:String) =>
  }

  Given("""^I have the following foods :$"""){ (table:DataTable) =>
    calorieCount = table.asMaps().map(_.values()).map(_.head.toDouble).fold(0.0)(_+_)
  }

  And("""^have eaten (.*) calories today""") { (calories:Double) =>
    assertEquals(calorieCount, calories)
  }
}

class ThenDefs extends ScalaDsl with EN {
  Then("""^I am "([^"]*)"$"""){ (arg0:String) =>
  }
}

File: java/src/test/java/cucumber/runtime/java/test/Stepdefs.java
package cucumber.runtime.java.test;

import cucumber.api.java.en.Given;

import java.util.List;

public class Stepdefs {
    @Given("I have {int} cukes in the belly")
    public void I_have_cukes_in_the_belly(int arg1) {
    }

    @Given("this data table:")
    public void this_data_table(List<Person> people) throws Throwable {
    }

    public static class Person {
        String first;
        String last;
    }
}
